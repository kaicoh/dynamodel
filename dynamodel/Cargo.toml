[package]
name = "dynamodel"
version = "0.1.0"
edition = "2021"

[dependencies]
dynamodel_derive = { path = "../dynamodel_derive" }
aws-sdk-dynamodb = { workspace = true }
thiserror = "1"
