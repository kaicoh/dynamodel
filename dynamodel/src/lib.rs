pub use dynamodel_derive::Dynamodel;

use aws_sdk_dynamodb::types::AttributeValue;
use std::num::{ParseFloatError, ParseIntError};
use thiserror::Error;

#[derive(Debug, Error)]
pub enum ConvertError {
    #[error("`{0}` field is not set")]
    FieldNotSet(String),

    #[error("expect `{0}` type, but got `{1:?}`")]
    AttributeValueUnmatched(String, AttributeValue),

    #[error("{0}")]
    ParseInt(#[from] ParseIntError),

    #[error("{0}")]
    ParseFloat(#[from] ParseFloatError),

    #[error(transparent)]
    Other(Box<dyn std::error::Error + Send + Sync>),
}
