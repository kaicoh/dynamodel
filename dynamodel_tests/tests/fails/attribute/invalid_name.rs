use dynamodel::Dynamodel;

#[derive(Dynamodel)]
struct InvalidAttributeName {
    #[dynamodel(foo(path = "unknown"))]
    bytes: Vec<u8>,
}

fn main() {}
