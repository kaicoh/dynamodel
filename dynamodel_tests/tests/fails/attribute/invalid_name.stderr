error: Unknown field: `foo`
 --> tests/fails/attribute/invalid_name.rs:5:17
  |
5 |     #[dynamodel(foo(path = "unknown"))]
  |                 ^^^
